app/src/test/java/uk/ac/ebi/atlas/search/species/SpeciesSearchDaoIT.java

```diff
@@ -0,0 +1,99 @@
+package uk.ac.ebi.atlas.search.species;
+
+import com.google.common.collect.ImmutableSet;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import uk.ac.ebi.atlas.configuration.TestConfig;
+import uk.ac.ebi.atlas.solr.cloud.SolrCloudCollectionProxyFactory;
+
+import javax.inject.Inject;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.ac.ebi.atlas.solr.bioentities.BioentityPropertyName.SYMBOL;
+
+@WebAppConfiguration
+@ExtendWith(SpringExtension.class)
+@ContextConfiguration(classes = TestConfig.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+class SpeciesSearchDaoIT {
+    @Inject
+    private SolrCloudCollectionProxyFactory collectionProxyFactory;
+
+    private SpeciesSearchDao subject;
+
+    @BeforeEach
+    void setUp() {
+        subject = new SpeciesSearchDao(collectionProxyFactory);
+    }
+
+    @Test
+    void whenEmptySearchTextProvidedReturnEmptyOptional() {
+        var searchText = "";
+        var category = SYMBOL.name;
+
+        var species = subject.searchSpecies(searchText, category);
+
+        assertThat(species.isEmpty()).isTrue();
+    }
+
+    @Test
+    void whenNoCategoryButValidGeneIdProvidedThenReturnResult() {
+        var searchText = "ACRV1";
+        String category = null;
+        var expectedSpecies = "Homo_sapiens";
+
+        var species = subject.searchSpecies(searchText, category);
+
+        assertThat(species.isPresent()).isTrue();
+        assertThat(species.get()).contains(expectedSpecies);
+    }
+
+    @Test
+    void whenNoDocumentsAreFoundInBioEntitiesReturnEmptyOptional() {
+        var species = subject.searchSpecies("FOOBAR", SYMBOL.name);
+
+        assertThat(species.isPresent()).isTrue();
+        assertThat(species.get()).isEmpty();
```

Comment:
Same as above. If you prefer you can use `hasValueSatisfying(Collection foo -> foo.isEmpty())`.